### PR TITLE
refactor: split foxy, yearn deposit and withdraw into multiple components

### LINKED ISSUE #1430: Refactor Yearn & Foxy Deposit and withdrawal files.
### Overview

The files containing the screens for yearn/foxy deposit and withdraw are hundreds of lines long. When written originally, yearn was written in one file to meet a deadline, foxy was copy/pasted based on that pattern.

Within these files, there are a large switch cases which use the route to determine which component for the screen to display.

For this bounty, we want to split the components and their functions within the switch cases into their own files. So the deposit view and its associated functions would live in it's own file named Deposit.tsx (see References and additional details for file structure) and so on for the reset of the components within the switch cases.

### References and additional details

Files can be found here:

src/features/defi/providers/yearn/components/YearnManager/Deposit/YearnDeposit.tsx
src/features/defi/providers/yearn/components/YearnManager/Withdraw/YearnWithdraw.tsx
src/features/defi/providers/foxy/components/FoxyManager/Deposit/FoxyDeposit.tsx
src/features/defi/providers/foxy/components/FoxyManager/Withdraw/FoxyWithdraw.tsx

Create four new directories all called components where the individual components will live:
src/features/defi/providers/yearn/components/YearnManager/Deposit/components/
Deposit.tsx
Approve.tsx
Confirm.tsx
Status.txs

src/features/defi/providers/yearn/components/YearnManager/Withdraw/components/
Withdraw.tsx
Confirm.tsx
Status.tsx

src/features/defi/providers/foxy/components/FoxyManager/Deposit/components/
Deposit.tsx
Approve.tsx
Confirm.tsx
Status.txs

src/features/defi/providers/foxy/components/FoxyManager/Withdraw/components/
Withdraw.tsx
Approve.tsx
Confirm.tsx
Status.txs

### Acceptance Criteria

- The user facing modal functionality is unchanged.
- The components used within the switch case are imported from their own files and only necessary props are sent into them.
- Functions isolated to one component live in the new file with the component.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

### Estimated effort

1 day

### Sponsor / Stakeholder

Discord: Theobold Frazier#5808 github: theoboldfrazier

### Bounty Hunters

- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info
- Please ***do not*** start work on this issue until you are approved in Gitcoin.

### PR DESCRIPTION
**Pull request by NeOMakinG**
## Description
- We should split every Foxy, Yearn deposit and withdraw into multiple components in order to simplify the router file
- Every file should have its own logic, parameters sent from the router are needed parameters only

More information inside the linked issue

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)
closes #1430

## Risk
This PR modifies the whole Yearn and Foxy modal scope, both features could break if I did some mistakes

## Testing
You should test the whole Foxy and Yean deposit/withdraw features

### CODE REVIEW

File: src/features/defi/providers/foxy/components/FoxyManager/Deposit/DepositReducer.ts [hunk 1]
```diff
@@ -104,7 +104,7 @@ type SetTxid = {
   payload: string
 }
 
-type FoxyDepositActions =
+export type FoxyDepositActions =
```
Comment (theoboldfrazier) [inline src/features/defi/providers/foxy/components/FoxyManager/Deposit/DepositReducer.ts:107]: Same, move to common

File: src/features/defi/providers/foxy/components/FoxyManager/Deposit/components/Approve.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import { poll } from 'lib/poll/poll'
+import { marketApi } from 'state/slices/marketDataSlice/marketDataSlice'
+import { selectAssetByCAIP19, selectMarketDataById } from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { DepositPath } from '../DepositCommon'
+import { DepositContext } from '../DepositContext'
+import { FoxyDepositActionType } from '../DepositReducer'
+
+export type FoxyApproveProps = {
```
Comment (theoboldfrazier) [inline src/features/defi/providers/foxy/components/FoxyManager/Deposit/components/Approve.tsx:24]: Remove export
```suggestion
type FoxyApproveProps = {
```

File: src/features/defi/providers/foxy/components/FoxyManager/Deposit/components/Confirm.tsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+import { poll } from 'lib/poll/poll'
+import { marketApi } from 'state/slices/marketDataSlice/marketDataSlice'
+import { selectAssetByCAIP19, selectMarketDataById } from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { DepositPath } from '../DepositCommon'
+import { DepositContext } from '../DepositContext'
+import { FoxyDepositActionType } from '../DepositReducer'
+
+export type FoxyConfirmProps = {
```
Comment (theoboldfrazier) [inline src/features/defi/providers/foxy/components/FoxyManager/Deposit/components/Confirm.tsx:28]:
```suggestion
type FoxyConfirmProps = {
```

File: src/features/defi/providers/foxy/components/FoxyManager/Deposit/components/Status.tsx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+import { Text } from 'components/Text'
+import { useBrowserRouter } from 'hooks/useBrowserRouter/useBrowserRouter'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { marketApi } from 'state/slices/marketDataSlice/marketDataSlice'
+import { selectAssetByCAIP19, selectMarketDataById } from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { DepositContext } from '../DepositContext'
+
+export type FoxyStatusProps = {
```
Comment (theoboldfrazier) [inline src/features/defi/providers/foxy/components/FoxyManager/Deposit/components/Status.tsx:23]:
```suggestion
type FoxyStatusProps = {
```

File: src/features/defi/providers/foxy/components/FoxyManager/Withdraw/WithdrawContext.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { createContext } from 'react'
+
+import { FoxyWithdrawActions, FoxyWithdrawState } from './WithdrawReducer'
+
+export interface IWithdrawContext {
```
Comment (theoboldfrazier) [inline src/features/defi/providers/foxy/components/FoxyManager/Withdraw/WithdrawContext.ts:5]:
```suggestion
interface IWithdrawContext {
```

File: src/features/defi/providers/foxy/components/FoxyManager/Withdraw/WithdrawReducer.ts [hunk 6]
```diff
@@ -110,7 +110,7 @@ type SetFoxyFee = {
   payload: string
 }
 
-type FoxyWithdrawActions =
+export type FoxyWithdrawActions =
```
Comment (theoboldfrazier) [inline src/features/defi/providers/foxy/components/FoxyManager/Withdraw/WithdrawReducer.ts:113]: Move to common

File: src/features/defi/providers/yearn/components/YearnManager/Deposit/DepositReducer.ts [hunk 7]
```diff
@@ -12,7 +12,7 @@ type YearnDepositValues = DepositValues &
     usedGasFee: string
   }
 
-type YearnDepositState = {
+export type YearnDepositState = {
```
Comment (theoboldfrazier) [inline src/features/defi/providers/yearn/components/YearnManager/Deposit/DepositReducer.ts:15]: Move to common

File: src/features/defi/providers/yearn/components/YearnManager/Deposit/DepositReducer.ts [hunk 8]
```diff
@@ -118,7 +118,7 @@ type SetTxid = {
   payload: string
 }
 
-type YearnDepositActions =
+export type YearnDepositActions =
```
Comment (theoboldfrazier) [inline src/features/defi/providers/yearn/components/YearnManager/Deposit/DepositReducer.ts:121]: Move to common

File: src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Approve.tsx [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+import { poll } from 'lib/poll/poll'
+import { marketApi } from 'state/slices/marketDataSlice/marketDataSlice'
+import { selectAssetByCAIP19, selectMarketDataById } from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { DepositPath } from '../DepositCommon'
+import { DepositContext } from '../DepositContext'
+import { YearnDepositActionType } from '../DepositReducer'
+
+export type YearnApproveProps = {
```
Comment (theoboldfrazier) [inline src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Approve.tsx:24]:
```suggestion
type YearnApproveProps = {
```

File: src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Confirm.tsx [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+import { poll } from 'lib/poll/poll'
+import { marketApi } from 'state/slices/marketDataSlice/marketDataSlice'
+import { selectAssetByCAIP19, selectMarketDataById } from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { DepositPath } from '../DepositCommon'
+import { DepositContext } from '../DepositContext'
+import { YearnDepositActionType } from '../DepositReducer'
+
+export type YearnConfirmProps = {
```
Comment (theoboldfrazier) [inline src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Confirm.tsx:28]:
```suggestion
type YearnConfirmProps = {
```

File: src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Deposit.tsx [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+  selectMarketDataById,
+  selectPortfolioCryptoBalanceByAssetId,
+} from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { DepositPath } from '../DepositCommon'
+import { DepositContext } from '../DepositContext'
+import { YearnDepositActionType } from '../DepositReducer'
+
+export type YearnDepositProps = {
```
Comment (theoboldfrazier) [inline src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Deposit.tsx:24]:
```suggestion
type YearnDepositProps = {
```

File: src/features/defi/providers/yearn/components/YearnManager/Withdraw/WithdrawContext.ts [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { createContext } from 'react'
+
+import { YearnWithdrawActions, YearnWithdrawState } from './WithdrawReducer'
+
+export interface IWithdrawContext {
```
Comment (theoboldfrazier) [inline src/features/defi/providers/yearn/components/YearnManager/Withdraw/WithdrawContext.ts:5]:
```suggestion
interface IWithdrawContext {
```

File: src/features/defi/providers/yearn/components/YearnManager/Withdraw/WithdrawReducer.ts [hunk 13]
```diff
@@ -12,7 +12,7 @@ type YearnWithdrawValues = WithdrawValues &
     usedGasFee: string
   }
 
-type YearnWithdrawState = {
+export type YearnWithdrawState = {
```
Comment (theoboldfrazier) [inline src/features/defi/providers/yearn/components/YearnManager/Withdraw/WithdrawReducer.ts:15]: Move to common

File: src/features/defi/providers/yearn/components/YearnManager/Withdraw/WithdrawReducer.ts [hunk 14]
```diff
@@ -113,7 +113,7 @@ type SetTxid = {
   payload: string
 }
 
-type YearnWithdrawActions =
+export type YearnWithdrawActions =
```
Comment (theoboldfrazier) [inline src/features/defi/providers/yearn/components/YearnManager/Withdraw/WithdrawReducer.ts:116]: Move to common

**Review by theoboldfrazier — COMMENTED** [review]
Overall, very well done. I pulled down and everything was working. I left a few comments to clean up a bit. Once those are addressed, I'll approve.

**Review by cjthompson — APPROVED** [review]

**Comment by cjthompson** [comment]
Can we get Ops to sign off on this ephemeral branch before we merge this?

**Comment by 0xdef1cafe** [comment]
@cjthompson just FYI - if we need testing from operations - we can just add the `needs operations` label, and make a thread in `#operations-publicchat` in discord and they'll get right on it

https://discord.com/channels/554694662431178782/876953436908822568/963153136569176074

**Review by theoboldfrazier — APPROVED** [review]
Looks great! Nice work.

**Pull request merged**